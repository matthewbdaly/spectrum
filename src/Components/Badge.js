import React, { Component } from 'react';
import './Badge.scss';

class Badge extends Component {
  constructor(props) {
    super(props);
    let theme = this.props.theme;
    if (this.props.inverse) {
      theme += " inverse";
    }
    this.state = {
      theme: theme
    };
  }
  render() {
    return (
      <span className={ 'badge' + (this.state.theme ? (' ' + this.state.theme) : '') }>
        {this.props.children}
      </span>
    );
  }
}

export default Badge;
