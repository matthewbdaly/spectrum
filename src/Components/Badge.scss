@import "../variables";

.badge {
    border-radius: 10px;
    background-color: lightgray;
    padding: 5px 14px;

    &.inverse {
        border-width: $btn-border-width;
        border-style: $border-style;
    }

    &.main {
        background-color: $theme-default;
        color: $background;
        border-color: $theme-default;

        &.inverse {
            color: $theme-default;
            background-color: $background;
        }
    }

    &.cool {
        background-color: $theme-cool;
        color: $background;
        border-color: $theme-cool;

        &.inverse {
            color: $theme-cool;
            background-color: $background;
        }
    }

    &.negative {
        background-color: $theme-hot;
        color: $background;
        border-color: $theme-hot;

        &.inverse {
            color: $theme-hot;
            background-color: $background;
        }
    }
}
