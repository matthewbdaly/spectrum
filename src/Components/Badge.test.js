import React from 'react';
import ReactDOM from 'react-dom';
import Badge from './Badge';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Badge />, div);
});

it('should match the snapshot', () => {
  const div = document.createElement('div');
  const el = ReactDOM.render(<Badge />, div);
  expect(el).toMatchSnapshot();
});
