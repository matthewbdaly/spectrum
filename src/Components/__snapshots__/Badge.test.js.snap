// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Badge {
  "_reactInternalFiber": FiberNode {
    "_debugID": 8,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "<PROJECT_ROOT>/src/Components/Badge.test.js",
      "lineNumber": 12,
    },
    "alternate": null,
    "child": FiberNode {
      "_debugID": 9,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "<PROJECT_ROOT>/src/Components/Badge.js",
        "lineNumber": 29,
      },
      "alternate": null,
      "child": null,
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": undefined,
        "className": "badge",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": undefined,
        "className": "badge",
      },
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <span
        class="badge"
/>,
      "tag": 5,
      "type": "span",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": Object {
      "theme": undefined,
    },
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugID": 6,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 6,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <span
                        class="badge"
            />
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "capturedValues": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Badge />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Badge />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 32,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Badge />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <span
                    class="badge"
          />
</div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <Badge />,
        },
        "callbackList": null,
        "capturedValues": null,
        "expirationTime": 0,
        "first": null,
        "hasForceUpdate": false,
        "isInitialized": true,
        "isProcessing": false,
        "last": null,
      },
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": Object {
    "theme": undefined,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
